### ISSUE TITLE
Customised annotation callout closes automatically when changing the viewport 

### ISSUE THREAD

**Issue by pushian** [mapbox/mapbox-gl-native#10919]
![example](https://user-images.githubusercontent.com/16488850/34884928-3f749e6e-f7f9-11e7-8ada-f5687880c565.gif)

**Platform: iOS**
**Mapbox SDK version:  "https://www.mapbox.com/ios-sdk/Mapbox-iOS-SDK.json" ~> 3.7**

### Steps to trigger behavior

 1. Tap on the annotation
 2. The customised callout view will appear
 3. Move the map slightly a bit and the callout view will disappear

### Expected behavior
The callout view should remain when the map is moving and should only be dismissed when the map is tapped somewhere else

### Actual behavior
Move the map slightly a bit and the callout view will disappear

**Comment by pushian** [comment]
@frederoni thanks for the information and I have tried to set as following:
`
class CustomCalloutView: UIView, MGLCalloutView {
var dismissesAutomatically = false
}
` 
It is working that the callout view will now remain when I am changing the viewport but the callout view is not moving together with the viewport. Is there any other settings that I have to implement to make the callout view moving also?
![example2](https://user-images.githubusercontent.com/16488850/34901630-e3880fa6-f847-11e7-9f52-9fd7cc6fb9fc.gif)

**Comment by pushian** [comment]
I tried to implement the variable isAnchoredToAnnotation but it resulted in a very strange behaviour. Each time I click on one of the annotation, the callout view will always appear at the top left corner of the screen (check out the gif I have uploaded). 
All the markers on the map are displayed using Features and Layers. I am using the following code to show the callout view:
`func showCallout(feature: MGLPointFeature) {`
        `let point = MGLPointFeature()`
        `point.title = feature.attributes["name"] as? String`
        `point.coordinate = feature.coordinate`
        `mapView.selectAnnotation(point, animated: true)`
    `}`
![example3](https://user-images.githubusercontent.com/16488850/34901797-3bde90c4-f84a-11e7-82ae-957ece031b23.gif)
